docs/parsers.rst

```diff
@@ -0,0 +1,126 @@
+- mapPriority,Integer,Within a map group, the mapPriority specifies the order in which complex map records should be evaluated to determine the correct map target.,
+- mapRule,String,A machine-readable rule, (evaluating to either ‘true’ or ‘false’ at run-time) that indicates whether this map record should be selected within its map group.,
+- mapAdvice,String,Human-readable advice that may be employed by the software vendor to give an end-user advice on selection of the appropriate target code. This includes a) a summary statement of the map rule logic, b) a statement of any limitations of the map record and c) additional classification guidance for the coding professional.,
+- mapTarget,String,The target ICD-10 classification code of the map record.,
+- correlationId,SctId,A child of |Map correlation value| in the metadata hierarchy, identifying the correlation between the SNOMED CT concept and the target code.,
+- mapCategoryId,SctId,Identifies the SNOMED CT concept in the metadata hierarchy which is the MapCategory for the associated map record. This is a subtype of 447634004 |ICD-10 Map Category value|.,
+
+Mappings: SSSOM::SNOMED_Complex_Map
+-----------------------------------
+Copy/pasta of state of mappings as of 2022/03/04:
+
+'subject_id': f'SNOMED:{row["referencedComponentId"]}',
+'subject_label': row['referencedComponentName'],
+
+# 'predicate_id': 'skos:exactMatch',
+# - mapCategoryId: can use for mapping predicate? Or is correlationId more suitable?
+#   or is there a SKOS predicate I can map to in case where predicate is unknown? I think most of these
+#   mappings are attempts at exact matches, but I can't be sure (at least not without using these fields
+#   to determine: mapGroup, mapPriority, mapRule, mapAdvice).
+# mapCategoryId,mapCategoryName: Only these in set: 447637006 "MAP SOURCE CONCEPT IS PROPERLY CLASSIFIED",
+#   447638001 "MAP SOURCE CONCEPT CANNOT BE CLASSIFIED WITH AVAILABLE DATA",
+#   447639009 "MAP OF SOURCE CONCEPT IS CONTEXT DEPENDENT"
+# 'predicate_modifier': '???',
+#   Description: Modifier for negating the prediate. See https://github.com/mapping-commons/sssom/issues/40
+#   Range: PredicateModifierEnum: (joe: only lists 'Not' as an option)
+#   Example: Not Negates the predicate, see documentation of predicate_modifier_enum
+# - predicate_id <- mapAdvice?
+# - predicate_modifier <- mapAdvice?
+#   mapAdvice: Pipe-delimited qualifiers. Ex:
+#   "ALWAYS Q71.30 | CONSIDER LATERALITY SPECIFICATION"
+#   "IF LISSENCEPHALY TYPE 3 FAMILIAL FETAL AKINESIA SEQUENCE SYNDROME CHOOSE Q04.3 | MAP OF SOURCE CONCEPT
+#   IS CONTEXT DEPENDENT"
+#   "MAP SOURCE CONCEPT CANNOT BE CLASSIFIED WITH AVAILABLE DATA"
+'predicate_id': f'SNOMED:{row["mapCategoryId"]}',
+'predicate_label': row['mapCategoryName'],
```

Comment:
Can you share an example of how the resulting sssom looks like, maybe, 10 rows in markdown here in this pull request?

Reply:
Yep, sure thing.

Reply:
|subject_id            |predicate_id    |object_id      |match_type |subject_label                               |subject_category|predicate_label                           |predicate_modifier|object_label                                        |object_category|author_id|author_label|reviewer_id|reviewer_label|creator_id|creator_label|license|subject_source|subject_source_version|object_source|object_source_version|mapping_provider|mapping_cardinality|mapping_tool|mapping_tool_version|mapping_date|confidence|subject_match_field|object_match_field|match_string|subject_preprocessing|object_preprocessing|match_term_type|semantic_similarity_score|semantic_similarity_measure|see_also|other                                                                                                                                                                                                                                                                                                                                                                                                                                                 |comment|
|----------------------|----------------|---------------|-----------|--------------------------------------------|----------------|------------------------------------------|------------------|----------------------------------------------------|---------------|---------|------------|-----------|--------------|----------|-------------|-------|--------------|----------------------|-------------|---------------------|----------------|-------------------|------------|--------------------|------------|----------|-------------------|------------------|------------|---------------------|--------------------|---------------|-------------------------|---------------------------|--------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------|
|SNOMED:10000006       |SNOMED:447637006|ICD10CM:R07.9  |Unspecified|Radiating chest pain                        |                |MAP SOURCE CONCEPT IS PROPERLY CLASSIFIED |                  |Chest pain, unspecified                             |               |         |            |           |              |          |             |       |              |                      |             |                     |                |                   |            |                    |2021-09-01  |          |                   |                  |            |                     |                    |               |                         |                           |        |id=4c4d2755-e65f-52dc-8353-ecf437495eb8&#124;active=1&#124;moduleId=5991000124107&#124;refsetId=6011000124106&#124;mapGroup=1&#124;mapPriority=1&#124;mapRule=TRUE&#124;mapAdvice=ALWAYS R07.9                                                                                                                                                                                                                                                                                           |       |
|SNOMED:10001005       |SNOMED:447639009|ICD10CM:P36.9  |Unspecified|Bacterial sepsis                            |                |MAP OF SOURCE CONCEPT IS CONTEXT DEPENDENT|                  |Bacterial sepsis of newborn, unspecified            |               |         |            |           |              |          |             |       |              |                      |             |                     |                |                   |            |                    |2021-09-01  |          |                   |                  |            |                     |                    |               |                         |                           |        |id=f7e1ed02-3107-5a6c-9439-21e9879fb746&#124;active=1&#124;moduleId=5991000124107&#124;refsetId=6011000124106&#124;mapGroup=1&#124;mapPriority=1&#124;mapRule=IFA 445518008 &#124; Age at onset of clinical finding (observable entity) &#124; < 29.0 days&#124;mapAdvice=IF AGE AT ONSET OF CLINICAL FINDING BEFORE 29.0 DAYS CHOOSE P36.9 &#124; CONSIDER ADDITIONAL CODE TO IDENTIFY SPECIFIC CONDITION OR DISEASE &#124; DESCENDANTS NOT EXHAUSTIVELY MAPPED &#124; MAP OF SOURCE CONCEPT IS CONTEXT DEPENDENT|       |
|SNOMED:10001005       |SNOMED:447637006|ICD10CM:A41.9  |Unspecified|Bacterial sepsis                            |                |MAP SOURCE CONCEPT IS PROPERLY CLASSIFIED |                  |Sepsis, unspecified organism                        |               |         |            |           |              |          |             |       |              |                      |             |                     |                |                   |            |                    |2021-09-01  |          |                   |                  |            |                     |                    |               |                         |                           |        |id=907aa2b0-a832-5254-acdf-6f366e406d2b&#124;active=1&#124;moduleId=5991000124107&#124;refsetId=6011000124106&#124;mapGroup=1&#124;mapPriority=2&#124;mapRule=OTHERWISE TRUE&#124;mapAdvice=ALWAYS A41.9 &#124; CONSIDER ADDITIONAL CODE TO IDENTIFY SPECIFIC CONDITION OR DISEASE &#124; DESCENDANTS NOT EXHAUSTIVELY MAPPED                                                                                                                                                                      |       |
|SNOMED:10007009       |SNOMED:447637006|ICD10CM:Q04.8  |Unspecified|Coffin-Siris syndrome                       |                |MAP SOURCE CONCEPT IS PROPERLY CLASSIFIED |                  |Other specified congenital malformations of brain   |               |         |            |           |              |          |             |       |              |                      |             |                     |                |                   |            |                    |2021-09-01  |          |                   |                  |            |                     |                    |               |                         |                           |        |id=0b918513-be22-5059-beb6-e8348dfc631f&#124;active=1&#124;moduleId=5991000124107&#124;refsetId=6011000124106&#124;mapGroup=1&#124;mapPriority=1&#124;mapRule=TRUE&#124;mapAdvice=ALWAYS Q04.8                                                                                                                                                                                                                                                                                           |       |
|SNOMED:1001000119102  |SNOMED:447637006|ICD10CM:I26.99 |Unspecified|Pulmonary embolism with pulmonary infarction|                |MAP SOURCE CONCEPT IS PROPERLY CLASSIFIED |                  |Other pulmonary embolism without acute cor pulmonale|               |         |            |           |              |          |             |       |              |                      |             |                     |                |                   |            |                    |2021-09-01  |          |                   |                  |            |                     |                    |               |                         |                           |        |id=7f3f0a95-0441-57ba-a58f-03d2f2736c60&#124;active=1&#124;moduleId=5991000124107&#124;refsetId=6011000124106&#124;mapGroup=1&#124;mapPriority=1&#124;mapRule=TRUE&#124;mapAdvice=ALWAYS I26.99                                                                                                                                                                                                                                                                                          |       |
|SNOMED:10017004       |SNOMED:447637006|ICD10CM:K03.0  |Unspecified|Occlusal wear of teeth                      |                |MAP SOURCE CONCEPT IS PROPERLY CLASSIFIED |                  |Excessive attrition of teeth                        |               |         |            |           |              |          |             |       |              |                      |             |                     |                |                   |            |                    |2021-09-01  |          |                   |                  |            |                     |                    |               |                         |                           |        |id=3fa74c85-56d0-5819-97e8-80c8d4c053c0&#124;active=1&#124;moduleId=5991000124107&#124;refsetId=6011000124106&#124;mapGroup=1&#124;mapPriority=1&#124;mapRule=TRUE&#124;mapAdvice=ALWAYS K03.0                                                                                                                                                                                                                                                                                           |       |
|SNOMED:100191000119105|SNOMED:447637006|ICD10CM:N42.89 |Unspecified|Asymmetry of prostate                       |                |MAP SOURCE CONCEPT IS PROPERLY CLASSIFIED |                  |Other specified disorders of prostate               |               |         |            |           |              |          |             |       |              |                      |             |                     |                |                   |            |                    |2021-09-01  |          |                   |                  |            |                     |                    |               |                         |                           |        |id=51537076-a3bf-59f8-aa5f-f571dfcf32cb&#124;active=1&#124;moduleId=5991000124107&#124;refsetId=6011000124106&#124;mapGroup=1&#124;mapPriority=1&#124;mapRule=TRUE&#124;mapAdvice=ALWAYS N42.89                                                                                                                                                                                                                                                                                          |       |
|SNOMED:100211000119106|SNOMED:447637006|ICD10CM:M62.830|Unspecified|Muscle spasm of thoracic back               |                |MAP SOURCE CONCEPT IS PROPERLY CLASSIFIED |                  |Muscle spasm of back                                |               |         |            |           |              |          |             |       |              |                      |             |                     |                |                   |            |                    |2021-09-01  |          |                   |                  |            |                     |                    |               |                         |                           |        |id=a0263a65-4056-5cf9-8caa-669afce20189&#124;active=1&#124;moduleId=5991000124107&#124;refsetId=6011000124106&#124;mapGroup=1&#124;mapPriority=1&#124;mapRule=TRUE&#124;mapAdvice=ALWAYS M62.830                                                                                                                                                                                                                                                                                         |       |
|SNOMED:1002229008     |SNOMED:447637006|ICD10CM:L75.2  |Unspecified|Apocrine miliaria of areola                 |                |MAP SOURCE CONCEPT IS PROPERLY CLASSIFIED |                  |Apocrine miliaria                                   |               |         |            |           |              |          |             |       |              |                      |             |                     |                |                   |            |                    |2021-09-01  |          |                   |                  |            |                     |                    |               |                         |                           |        |id=65fa44e5-272d-54e5-b787-cb977fc24720&#124;active=1&#124;moduleId=5991000124107&#124;refsetId=6011000124106&#124;mapGroup=1&#124;mapPriority=1&#124;mapRule=TRUE&#124;mapAdvice=ALWAYS L75.2                                                                                                                                                                                                                                                                                           |       |
|SNOMED:1002253002     |SNOMED:447637006|ICD10CM:Z28.3  |Unspecified|Immunization series incomplete              |                |MAP SOURCE CONCEPT IS PROPERLY CLASSIFIED |                  |Underimmunization status                            |               |         |            |           |              |          |             |       |              |                      |             |                     |                |                   |            |                    |2021-09-01  |          |                   |                  |            |                     |                    |               |                         |                           |        |id=c2ab86f4-29c5-5ea3-8818-1e389d9c59a6&#124;active=1&#124;moduleId=5991000124107&#124;refsetId=6011000124106&#124;mapGroup=1&#124;mapPriority=1&#124;mapRule=TRUE&#124;mapAdvice=ALWAYS Z28.3 &#124; CONSIDER ADDITIONAL CODE TO IDENTIFY SPECIFIC CONDITION OR DISEASE                                                                                                                                                                                                                      |       |

Reply:
- match_type does not exist anymore
- Why all these empty columns?
- predicate_id is not legal - I looked this SNOMED rel up but could not understand what it means at all. Can you explain how "Map source concept is properly classified" is a mapping relation? how does it map to skos?
- you should add a test that runs `validate_file` on the a small sample of the output I think,

Reply:
* `match_type`: Oh this is an old copy. I haven't re-ran it in awhile. The code has since been updated to `mapping_justification`.
* Empty columns: I haven't determined how to populate these SSSOM fields. For many of them, there may not be any information available. I have enumerated the SSSOM Mapping data model as of 2022/30 in my code comments ([link](https://github.com/mapping-commons/sssom-py/blob/snomed-icd-mappings-work/sssom/parsers.py#L825)). If you click that link and check it out, you'll see that I have some thoughts/questions as to how to populate. Any feedback/suggestions to those questions/comments are appreciated.
* `predicate_id`: I opened a [review comment](https://github.com/mapping-commons/sssom-py/pull/207#discussion_r819104571) in the first commit about populating this field from SNOMED's `mapCategoryId` and `mapCategoryName` fields. They don't use too many mapping predicates, unfortunately. In that linked review comment, I stated that there were only 3 values in the field, but the ICD10CM mapping dataset only includes 1: "MAP SOURCE CONCEPT IS PROPERLY CLASSIFIED" (447637006). You might be able to find more information in the "codebook" that SNOMED has published on these mappings. That codebook is named "doc_Icd10cmMapReleaseNotes_Current-en-US_US1000124_20210901.pdf" which I have zipped up for you (w/ some other related files) here: 
[SNOMED ICD10CM map technical guide etc.zip](https://github.com/mapping-commons/sssom-py/files/9122043/SNOMED.ICD10CM.map.technical.guide.etc.zip)
* test: Sure thing, I'll add one.

Reply:
@joeflack4 TODO: Note to self: Add a test that does `validate_file` on the output.